### PR TITLE
feat: contact information

### PR DESCRIPTION
**Pull request by Graborg**
**Ticket:** [#119](https://trello.com/c/IAnS5cV9/119-kontaktuppgifter)
**Intent:** Adds the view for adding firstname/lastname and contact information

**How to test (optional):**

### All of the following commands should pass.

- [x] `npm test`
- [x] `npm run lint`
- [x] `npm run cypress`

### Browser testing

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] IE11
- [ ] Edge

### CODE REVIEW

**Comment by Graborg** [comment]
WIP

File: src/graphql/resolvers/mutations/updateContactInformation.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+  }
+`
+
+export const updateContactInformation = (
+  _: any,
+  { data }: { data: Contact },
+  { cache }: { cache: InMemoryCache }
+): Contact => {
+  const { contact } = cache.readQuery<{
```
Comment (lundeeell) [inline src/graphql/resolvers/mutations/updateContactInformation.ts]: Why a readQuery?

File: src/utils/helpers.ts [hunk 2]
```diff
@@ -49,7 +49,18 @@ interface WhoAmI extends StorageEntryProps {
...
+interface ContactInformation extends StorageEntryProps {
+  type: 'contact'
+  data: Contact
+}
+
+export type StorageEntry =
+  | Educations
+  | Occupations
+  | Skills
+  | ContactInformation
```
Comment (lundeeell) [inline src/utils/helpers.ts]: Myyy eeeeeeyyeyesss, breaking a perfect asc sorting 😢

File: src/utils/helpers.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import Cookies from 'js-cookie'
-import { LocalStateProps } from '../graphql/client'
+import { LocalStateProps, Contact } from '../graphql/client'
```
Comment (lundeeell) [inline src/utils/helpers.ts]: Cant we use generated contact?

File: src/graphql/client.ts [hunk 4]
```diff
@@ -109,13 +108,21 @@ const terminatingLink = split(
   authLink.concat(httpLink)
 )
 
+export type Contact = {
```
Comment (lundeeell) [inline src/graphql/client.ts]: Why cant we use the generated, like we do for all other types

File: src/graphql/client.ts [hunk 5]
```diff
@@ -109,13 +108,21 @@ const terminatingLink = split(
...
+  name: string
+  email: string
+  telephone: string
```
Comment (lundeeell) [inline src/graphql/client.ts]: Here, we treat telephone as a string. Int the API, it is an int!

File: cypress/integration/Start/AddContactInfo.feature [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+Feature: AddContactInfo
+
+  User can add contact information
+
+  Background:
+    Given I open the kontakt-page
+
+  Scenario: User can add contact information
+    When I type "Gordon Freeman" as "name"
+    And I type "gordon@freeman.com" as "email"
```
[the remark this answers is not in the document]
Comment (momentiris) [reply cypress/integration/Start/AddContactInfo.feature]: good point!

File: src/utils/helpers.ts (hunk 2 above)
Comment (momentiris) [reply src/utils/helpers.ts]: Haha, sure. my linting does not tell me so I didnt think about it

**Pull request merged**